src/main/scala/uk/gov/hmrc/mongo/json/ExtraBSONHandlers.scala

```diff
@@ -10,7 +10,7 @@ trait ExtraBSONHandlers {
     new BSONDocumentReader[Map[String, T]] {
     def read(doc: BSONDocument): Map[String, T] = {
       doc.elements.collect {
-        case (key, value) => value.seeAsOpt[T](reader) map {
+        case BSONElement(key, value) => value.seeAsOpt[T](reader) map {
```

Comment:
`elements` are now a collection of `BSONElement` rather than tuples. This does not effect return type of this function.